### DISCUSSION TITLE
How to write RadioButton / CheckBox / A.Option A B. Option B

### DISCUSSION THREAD

**Discussion by a44281071** [QuestPDF/QuestPDF#372] [Q&A] [answered]
RadioButton 
```` 
〇 Option A         ◉ Option B        〇 Option C        〇 Option D
````

CheckBox 
````
☐ Option A         ☑ Option B        ☐ Option C        ☐ Option D
````

Options
````
A. Option A         B. Option B        C. Option C        D. Option D
````

**Comment by girlpunk** [comment]
https://www.questpdf.com/api-reference/row

**Comment by MarcinZiabek** [comment]
For rendering shapes (checkbox / radio-button with various states), you can use the `Canvas` element to draw paths. https://materialdesignicons.com/ has thousands of icons - for each, you can access SVG path.

For example, this image:
![image](https://user-images.githubusercontent.com/9263853/195052292-4d8e4fbb-3bc8-4214-a546-da283697197c.png)

Can be repsresented by the following SVG path:
```
M18 16H14V18H18V20L21 17L18 14V16M11 4C8.8 4 7 5.8 7 8S8.8 12 11 12 15 10.2 15 8 13.2 4 11 4M11 14C6.6 14 3 15.8 3 18V20H12.5C12.2 19.2 12 18.4 12 17.5C12 16.3 12.3 15.2 12.9 14.1C12.3 14.1 11.7 14 11 14
```

**Reply by a44281071** [reply]
Any demos? I don't know how to display image with path code (not image file).

**Reply by MarcinZiabek** [reply]
I am using this approach in the QuestPDF Previewer to display a nice icon when a rendering exception happens. Please take a look here: https://github.com/QuestPDF/QuestPDF/blob/main/QuestPDF/Previewer/ExceptionDocument.cs#L45-L58

This code uses standard SkiaSharp API, feel free to read more in the official SkiaSharp documentation 😁

**Reply by a44281071** [reply]
Thanks. But it is not Image(),

**Reply by Vashdastampedeee** [reply]
The link is not working @MarcinZiabek is there possibility for me to have access and view this? I need this in my project.

**Reply by MarcinZiabek** [reply, marked as answer]
You can view the legacy implementation here: https://github.com/QuestPDF/QuestPDF/blob/2022.12.X-support/Source/QuestPDF/Previewer/ExceptionDocument.cs

However, for a modern approach, I suggest using SVG: 
https://www.questpdf.com/api-reference/image/svg.html
https://www.questpdf.com/api-reference/image/dynamic.html
https://www.questpdf.com/api-reference/skiasharp-integration.html

**Comment by Vashdastampedeee** [comment]
Hello brother thanks for sending the link it will be helpful to my project.
I really appreciated it.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
